src/main/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResource.java

```diff
@@ -64,4 +78,42 @@ public Response matchDataUse(final DataUseMatchPair matchPair) {
         }
     }
 
+  /**
+   * Most recent version of matching logic as implemented in FireCloud
+   *
+   * @param matchPair The DataUseMatchPairV3
+   * @return Response
+   */
+  @Path("/v3")
+  @POST
+  public Response matchDataUseV3(final DataUseMatchPairV3 matchPair) {
+    DataUseV3 purpose = matchPair.getPurpose();
+    DataUseV3 dataset = matchPair.getConsent();
+    try {
+      if (purpose != null && dataset != null) {
+        MatchResult matchResult = dataUseMatcherV3.matchPurposeAndDatasetV3(purpose, dataset);
+        MatchResultType match = matchResult.getMatchResultType();
+        List<String> failures = DataUseUtil.sanitize(matchResult.getMessage());
+        return Response
+            .ok()
+            .entity(ImmutableMap.of(
+                "result", match,
+                "matchPair", matchPair,
+                "failureReasons", failures))
+            .type(MediaType.APPLICATION_JSON)
+            .build();
+      }
```

Comment:
I think semgrep is complaining about using `matchPair` on line 101. Maybe if you had a string as input, sanitized that string, then deserialized it with Gson, it might resolve the complaint.

Reply:
I did a test and I was wrong, that doesn't solve the problem. I suggest adding `// nosemgrep` just before the return statement